File: clients/fsharp/Model/MinMaxRangeInt64.fs
#nowarn "0058"

namespace Komus24
namespace Komus24.Model

open Komus24

/// Range of values
type MinMaxRangeInt64 = {
    /// Minimal value
    Min: int64;
    /// Maximal  value
    Max: int64;
} with

    /// Write MinMaxRangeInt64 to writer
    member this.writeTo(writer: System.IO.BinaryWriter) =
        writer.Write this.Min
        writer.Write this.Max
        ()

    /// Read MinMaxRangeInt64 from reader
    static member readFrom(reader: System.IO.BinaryReader) = {
        Min = reader.ReadInt64()
        Max = reader.ReadInt64()
    }

File: clients/haskell/src/Model/MinMaxRangeInt64.hs
module Model.MinMaxRangeInt64 where

import Prelude hiding (id)
import qualified Trans
import Trans (Trans)
import Data.Int

-- | Range of values
data MinMaxRangeInt64 = MinMaxRangeInt64 {
    -- | Minimal value
    min :: Int64,
    -- | Maximal  value
    max :: Int64 }
    deriving Show

instance Trans MinMaxRangeInt64 where
    read = do
        min <- Trans.read
        max <- Trans.read
        return MinMaxRangeInt64 {
            min,
            max }
    
    write MinMaxRangeInt64 {
        min,
        max } = do
            Trans.write min
            Trans.write max